comp/core/sysprobeconfig/component.go

```diff
@@ -30,5 +33,15 @@ type Component interface {
 	Warnings() *config.Warnings
 
 	// SysProbeObject returns the wrapper sysconfig
-	SysProbeObject() *sysconfig.Config
+	SysProbeObject() *sysconfigtypes.Config
+}
+
+// NewNoneOptional return a None optional type for sysprobeconfig.Component.
+//
+// This helper allows code that needs a disabled Optional type for sysprobeconfig to get it. The helper is split from
+// the implementation to avoid linking with the dependencies from sysprobeconfig.
+func NewNoneOptional() fxutil.Module {
```

Comment:
As this is the counterpart of `Module()` what do you think about naming this function `NoneModule`?